### PR TITLE
Component event

### PR DESCRIPTION
**Pull request by tomrf1**
## What does this change?
Snippet atoms currently record 3 types of interaction in ophan:
- in view
- expand
- like/dislike

This currently uses the old `interaction` model in ophan. We want to migrate to the new `componentEvent` [model](https://github.com/guardian/ophan/blob/65caa8b5f4681a368dbf7d0e0dc8254525f39609/event-model/src/main/thrift/componentevent.thrift#L142).
For now both events will be sent to ophan, but the old method should be removed later.

## What is the value of this and can you measure success?
We want to do all tracking using the new model. We will be using this model for a dashboard.

## Tested in CODE?
Yes

### CODE REVIEW

File: static/src/javascripts/projects/journalism/snippet-feedback.js [hunk 1]
```diff
@@ -10,6 +10,28 @@ const SnippetFeedback = (options: { scroll: boolean } = { scroll: true }) => {
...
+    const sendOldStyleInteraction = (snippetId, component, value) => {
+        ophan.record({
+            atomId: snippetId,
+            component,
+            value,
+        });
+    };
+
+    const sendNewStyleInteraction = (snippetId, snippetType, action) => {
+        ophan.record({
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/journalism/snippet-feedback.js]: You will get type check by using [`submitComponentEvent`](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/commercial/acquisitions-ophan.js) instead.

**Comment by mchv** [comment]
Seems good to me, @regiskuckaertz do you think is fine as well?

File: static/src/javascripts/projects/journalism/snippet-feedback.js [hunk 2]
```diff
@@ -10,6 +10,28 @@ const SnippetFeedback = (options: { scroll: boolean } = { scroll: true }) => {
...
+            componentEvent: {
+                component: {
+                    componentType: `${snippetType.toUpperCase()}_ATOM`,
+                    id: snippetId,
+                    products: [],
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/journalism/snippet-feedback.js]: You can just leave these two out.

File: static/src/javascripts/projects/journalism/snippet-feedback.js (hunk 1 above)
Comment (tomrf1) [reply static/src/javascripts/projects/journalism/snippet-feedback.js]: ah, thanks!

**Comment by regiskuckaertz** [comment]
The new interaction call does not record in which variant the current reader is. Shouldn't this be accounted for?

**Review by regiskuckaertz — APPROVED** [review]
Boom 💥

**Pull request merged**